tests/system/libraries/ChromeLib.py

```diff
@@ -103,76 +152,89 @@ def _writeTestFile(testCase) -> str:
-		This may mean that there is a separator between document name and application name.
-		E.G. "htmlTest   Google Chrome", "html – Google Chrome" or perhaps no application name at all.
-		Rather than try to get this right, just use the doc title.
-		If this continues to be unreliable we could use Selenium or similar to start chrome and inform us
-		when it is ready.
+		spy.emulateKeyPress('control+F6')  # focus web content, chrome shortcut.
+		spy.wait_for_speech_to_finish()
+		afterControlF6Speech = spy.get_speech_at_index_until_now(addressSpeechIndex)
+		if f"document\n{ChromeLib._beforeMarker}" not in afterControlF6Speech:
+			builtIn.log(afterControlF6Speech, level="DEBUG")
+			return False
+
+		spy.emulateKeyPress('control+home')  # ensure we start at the top of the document
+		controlHomeSpeechIndex = spy.get_last_speech_index()
+		spy.emulateKeyPress('numpad8')  # report current line
+		spy.wait_for_speech_to_finish()
+		afterNumPad8Speech = spy.get_speech_at_index_until_now(controlHomeSpeechIndex)
+		if ChromeLib._beforeMarker not in afterNumPad8Speech:
+			builtIn.log(afterNumPad8Speech, level="DEBUG")
+			return False
+		return True
+
+	def toggleFocusChrome(self):
+		"""Remove focus, then refocus chrome
+		Attempt to work around NVDA missing focus / foreground events when chrome first opens
+		@returns: last speech index before refocusing chrome
 		"""
-		success, _success = _blockUntilConditionMet(
-			getValue=lambda: SetForegroundWindow(startsWithTestCaseTitle, builtIn.log),
+		spy = _NvdaLib.getSpyLib()
+		spy.emulateKeyPress('windows+d')
+		_blockUntilConditionMet(
 			giveUpAfterSeconds=5,
-			intervalBetweenSeconds=0.2
+			getValue=GetForegroundWindowTitle,
+			shouldStopEvaluator=lambda _title: self.chromeWindow.title != _title,
+			errorMessage="Chrome didn't lose focus"
 		)
-		if success:
-			return
-		windowInformation = ""
-		try:
-			windowInformation = f"Foreground Window: {GetForegroundWindowTitle()}.\n"
-			windowInformation += f"Open Windows: {GetVisibleWindowTitles()}"
-		except OSError as e:
-			builtIn.log(f"Couldn't retrieve active window information.\nException: {e}")
-		raise AssertionError(
-			"Unable to focus Chrome.\n"
-			f"{windowInformation}"
+		spy.emulateKeyPress('alt+tab')
+		_blockUntilConditionMet(
+			giveUpAfterSeconds=5,
+			getValue=GetForegroundWindowTitle,
+			shouldStopEvaluator=lambda _title: self.chromeWindow.title == _title,
+			errorMessage="Chrome didn't gain focus"
 		)
+		spy.wait_for_speech_to_finish()
+
+	def ensureChromeTitleCanBeReported(self, applicationTitle) -> int:
+		spy = _NvdaLib.getSpyLib()
+		afterFocusToggleIndex = spy.get_last_speech_index()
+		spy.emulateKeyPress('NVDA+t')
+		appTitleIndex = spy.wait_for_specific_speech(applicationTitle, afterIndex=afterFocusToggleIndex)
+		return appTitleIndex
 
-	def prepareChrome(self, testCase: str) -> None:
+	def prepareChrome(self, testCase: str, _doToggleFocus=False) -> None:

```

Comment:
```suggestion
	def prepareChrome(self, testCase: str, _doToggleFocus: bool = False) -> None:
```

Reply:
Do parameters with default values need explicit typing?

Reply:
I believe so, we generally do this as a variety of types may be supplied regardless of default value.